sdk-ext-lib/src/main/java/cash/z/ecc/sdk/model/WalletAddresses.kt

```diff
@@ -0,0 +1,51 @@
+package cash.z.ecc.sdk.model
+
+import cash.z.ecc.android.bip39.Mnemonics
+import cash.z.ecc.android.bip39.toSeed
+import cash.z.ecc.android.sdk.tool.DerivationTool
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+
+data class WalletAddresses(
+    val unified: String,
+    val shieldedOrchard: String,
+    val shieldedSapling: String,
+    val transparent: String,
+    val viewingKey: String
+) {
+    // Override to prevent leaking details in logs
+    override fun toString() = "WalletAddresses"
+
+    companion object {
+        suspend fun new(persistableWallet: PersistableWallet): WalletAddresses {
+            // Dispatcher needed because SecureRandom is loaded, which is slow and performs IO
+            // https://github.com/zcash/kotlin-bip39/issues/13
+            val bip39Seed = withContext(Dispatchers.IO) {
+                Mnemonics.MnemonicCode(persistableWallet.seedPhrase.joinToString()).toSeed()
+            }
+
+            // Dispatchers needed until an SDK is published with the implementation of
+            // https://github.com/zcash/zcash-android-wallet-sdk/issues/269
+            val viewingKey = withContext(Dispatchers.IO) {
```

Comment:
@pacu Can you help me confirm that for the addresses supported by the SDK, that I got the right values here?

There are methods for both `deriveUnifiedViewingKeys` and `deriveViewingKeys` for example, and it isn't clear to me which is the right one to call in this situation.

https://github.com/zcash/zcash-android-wallet-sdk/blob/6968797092487b70fe5629a4a0ad11064715d08f/sdk-lib/src/main/java/cash/z/ecc/android/sdk/tool/DerivationTool.kt

Reply:
Well for this situation you would have to access every component individually. 

The concept of UVKs that we have on the SDK now is different as the one that it's going to have later. Then there were no UVKs we actually kind of coined that term ourselves by grouping an IVK and a transparent public key.